### PR TITLE
Add PrimaryWithCriteria, OccurrenceForPrimary filters to the OpenAPI spec.

### PR DESCRIPTION
**Pull request by marikomedlock**
The filter implementations + tests were part of earlier PRs. This just exposes the filters to the UI, and in the future, criteria plugins.

### CODE REVIEW

File: service/src/main/resources/api/service_openapi.yaml [hunk 1]
```diff
@@ -1537,6 +1541,53 @@ components:
...
+        If the group_by_count_operator is set, then the group_by_count_value must also be set.
+        The group_by_count_attributes are optional. If undefined, we group by the id only.
+        When group_by fields are set, we filter on # of related occurrence instances.
+        e.g. filter person instances with >2 related condition_occurrence instances
+      properties:
+        entityGroup:
+          type: string
+        criteriaSubfilter:
+          $ref: "#/components/schemas/Filter"
+        occurrenceSubfiltersAndGroupByAttributes:
```
Comment (tjennison-google) [inline service/src/main/resources/api/service_openapi.yaml:1573]: What is the purpose of this?

File: service/src/main/resources/api/service_openapi.yaml [hunk 2]
```diff
@@ -1537,6 +1541,53 @@ components:
...
+        primarySubfilter:
+          $ref: "#/components/schemas/Filter"
+
+    PrimaryWithCriteriaFilter:
+      type: object
+      required: [entityGroup]
+      description: |
+        Filter the primary entity with a criteria-occurrence entity group (e.g. people where 
+        condition=diabetes and visit_type=inpatient).
+        If the group_by_count_operator is set, then the group_by_count_value must also be set.
```
Comment (tjennison-google) [inline service/src/main/resources/api/service_openapi.yaml:1564]: You should probably use the camelCase names (e.g. `groupByCountOperator`) to match the names in the file.

File: service/src/main/resources/api/service_openapi.yaml (hunk 1 above)
Comment (marikomedlock) [reply service/src/main/resources/api/service_openapi.yaml:1573]: It's for modifiers (e.g. visit type, age at occurrence) and group by attributes (e.g. start_date, date).

Comment (tjennison-google) [reply service/src/main/resources/api/service_openapi.yaml:1573]: Why is in a separate object though? I guess maybe the keys would be for each occurrence entity?

Comment (marikomedlock) [reply service/src/main/resources/api/service_openapi.yaml:1573]: Right, this allows for different sub-filters and group by attributes per occurrence entity. I'll add a description to make it clear the keys are the occurrence entity names.

**Review by tjennison-google — APPROVED** [review]

**Pull request merged**